tools/dnstester/dnstester.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"log"
+
+	"github.com/miekg/dns"
+)
+
+func main() {
+	dns.Handle(".", dns.HandlerFunc(func(w dns.ResponseWriter, r *dns.Msg) {
+		m := new(dns.Msg)
+		m.SetReply(r)
+
+		var rr dns.RR
+		var err error
+		switch r.Question[0].Qtype {
+		case dns.TypeA:
+			rr, err = dns.NewRR("inspektor-gadget.io. A 127.0.0.1")
```

Comment:
nit: I'd use a fake domain to make it clear we're not performing a real resolution.